crates/common/tedge_config/src/tedge_config_dto.rs

```diff
@@ -99,6 +102,15 @@ pub(crate) struct MqttConfigDto {
     pub(crate) external_keyfile: Option<FilePath>,
 }
 
+#[derive(Debug, Default, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct HttpConfigDto {
+    pub(crate) port: Option<u16>,
+    pub(crate) bind_address: Option<IpAddress>,
+    pub(crate) external_port: Option<u16>,
+    pub(crate) external_bind_address: Option<IpAddress>,
```

Comment:
@didier-wenzek, maybe i should have just kept port here, but I had in mind the scenario where we might want a http external address and mqtt on an internal one, or vice-versa

Reply:
Sounds fine

Reply:
If you add an `http.bind_address` here, you have also to add an implementation for `ConfigSettingAccessor<HttptBindAddressSetting>` and to use it in the mapper and the c8y_configuration plugin.

Note that the implementation of `ConfigSettingAccessor<HttptBindAddressSetting>::query()` is a convenient place to implement the logic defaulting to the MQTT setting when nothings specific is set for HTTP.